activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/KahaDBTransactionStore.java

```diff
@@ -319,7 +282,7 @@ public void commit(TransactionId txid, boolean wasPrepared, final Runnable preCo
      */
     @Override
     public void rollback(TransactionId txid) throws IOException {
-        if (txid.isXATransaction() || theStore.isConcurrentStoreAndDispatchTransactions() == false) {
+        if (txid.isXATransaction()) {
```

Comment:
I think you're actually changing the way this stuff works here? 

concurrentStoreAndDispatchTransactions is always false, so this OR statement is always going to be true regardless of if it's an XA transaction or not: 
`if (txid.isXATransaction() || theStore.isConcurrentStoreAndDispatchTransactions() == false)`

I think you actually want to delete the else, and remove the if all together?

Reply:
You are right. Thanks for noticing. Will update it shortly

Reply:
Fixed